tests/cpp/dlsym/dlr_dlsym_test.cc

```diff
@@ -0,0 +1,284 @@
+  EXPECT_EQ(num_outputs, 2);
+}
+
+TEST(DLR, TestGetDLROutputType) {
+  auto model = GetDLRModel();
+  // output 0
+  const char* output_type0;
+  EXPECT_EQ(GetDLROutputType(&model, 0, &output_type0), 0);
+  EXPECT_STREQ(output_type0, "int32");
+  // output 1
+  const char* output_type1;
+  EXPECT_EQ(GetDLROutputType(&model, 1, &output_type1), 0);
+  EXPECT_STREQ(output_type1, "float32");
+  DeleteDLRModel(&model);
+}
+
+TEST(DLR, GetDLRHasMetadata) {
+  auto model = GetDLRModel();
+  bool has_metadata;
+  EXPECT_EQ(GetDLRHasMetadata(&model, &has_metadata), 0);
+  EXPECT_TRUE(has_metadata);
+  DeleteDLRModel(&model);
+}
+
+TEST(DLR, TestGetDLROutputName) {
+  auto model = GetDLRModel();
+  // output 0
+  const char* output_name0;
+  EXPECT_EQ(GetDLROutputName(&model, 0, &output_name0), 0);
+  EXPECT_STREQ(output_name0, "ArgMax:0");
+  // output 1
+  const char* output_name1;
+  EXPECT_EQ(GetDLROutputName(&model, 1, &output_name1), 0);
+  EXPECT_STREQ(output_name1, "softmax_tensor:0");
+  DeleteDLRModel(&model);
+}
+
+TEST(DLR, TestGetDLROutputIndex) {
+  auto model = GetDLRModel();
+  // output 0
+  int output_index0;
+  EXPECT_EQ(GetDLROutputIndex(&model, "ArgMax:0", &output_index0), 0);
+  EXPECT_EQ(output_index0, 0);
+  // output 1
+  int output_index1;
+  EXPECT_EQ(GetDLROutputIndex(&model, "softmax_tensor:0", &output_index1), 0);
+  EXPECT_EQ(output_index1, 1);
+  DeleteDLRModel(&model);
+}
+
+TEST(DLR, TestGetDLRBackend) {
+  auto model = GetDLRModel();
+  const char* backend;
+  EXPECT_EQ(GetDLRBackend(&model, &backend), 0);
+  EXPECT_STREQ(backend, "tvm");
+  DeleteDLRModel(&model);
+}
+
+TEST(DLR, TestRunDLRModel_GetDLROutput) {
+  auto model = GetDLRModel();
+  size_t img_size = 224*224*3;
+  float* img = LoadImageAndPreprocess("cat224-3.txt", img_size, 1);
+  int64_t shape[4] = {1, 224, 224, 3};
+  const char* input_name = "input_tensor";
+  EXPECT_EQ(SetDLRInput(&model, input_name, shape, img, 4), 0);
+  EXPECT_EQ(RunDLRModel(&model), 0);
+  // output 0
+  int output0[1];
+  EXPECT_EQ(GetDLROutput(&model, 0, output0), 0);
+  EXPECT_EQ(output0[0], 112);
+  EXPECT_EQ(GetDLROutputByName(&model, "ArgMax:0", output0), 0);
+  EXPECT_EQ(output0[0], 112);
+  const void* output0_p;
+  EXPECT_EQ(GetDLROutputPtr(&model, 0, &output0_p), 0);
+  EXPECT_EQ(((int*)output0_p)[0], 112);
+  // output 1
+  float output1[1001];
+  EXPECT_EQ(GetDLROutput(&model, 1, output1), 0);
+  EXPECT_GT(output1[112], 0.01);
+  EXPECT_EQ(GetDLROutputByName(&model, "softmax_tensor:0", output1), 0);
+  EXPECT_GT(output1[112], 0.01);
+  float* output1_p;
+  EXPECT_EQ(GetDLROutputPtr(&model, 1, (const void**) &output1_p), 0);
+  EXPECT_GT(output1_p[112], 0.01);
+  for (int i = 0; i < 1001; i++) {
+    EXPECT_EQ(output1_p[i], output1[i]);
+  }
+  delete [] img;
+  DeleteDLRModel(&model);
+}
+
+
+int main(int argc, char** argv) {
+  testing::InitGoogleTest(&argc, argv);
+  void* dlr = dlopen("lib/libdlr.so", RTLD_LAZY | RTLD_LOCAL);
```

Comment:
Is there a way we can move this setup to a gtest fixture or just a helper function?

Reply:
added InitDLR